src/natural_language_geocoding/models.py

```diff
@@ -58,10 +58,51 @@ class CoastOf(SpatialNodeType):
     child_node: "AnySpatialNodeType"
 
     def to_geometry(self, place_lookup: PlaceLookup) -> BaseGeometry | None:
-        child_bounds = self.child_node.to_geometry(place_lookup)
-        if child_bounds is None:
+        if child_bounds := self.child_node.to_geometry(place_lookup):
+            return coastline_of(child_bounds)
+        return None
+
+
+# The number of kilometers of buffer added to each shape to ensure that the areas that are very
+# close to intersection do intersect when computing border collisions
+_BORDER_BUFFER_SIZE = 3.5
+
+
+class BorderBetween(SpatialNodeType):
+    """Represents the border of two areas."""
```

Comment:
Recommend describing the operation in a bit more detail with examples and including the kind of geometry it produces.